src/neptune_mlflow_exporter/impl/utils.py

```diff
@@ -0,0 +1,91 @@
+__all__ = [
+    "ArtifactUploadStrategy",
+    "FileUploadStrategy",
+    "DirectoryUploadStrategy",
+]
+
+import os
+import shutil
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from pathlib import Path
+
+from mlflow.artifacts import download_artifacts
+from mlflow.entities import FileInfo
+
+try:
+    from neptune import Run
+except ImportError:
+    from neptune.new.metadata_containers import Run
+
+
+class ArtifactUploadStrategy(ABC):
```

Comment:
This could be simplified by a lot. The logic is much more common for both strategies. In general:
- Ensure the size is ok - No need to differentiate as it could be generalized
- Download artifact (could be downloaded to some "temp" directory to simplify) - No need to differentiate
- Upload file/files (this is the only part that I think needs to depend on artifact type)
- Clean the "temp" directory with `shutil.rmtree` - No need to differentiate